partner_firstname/migrations/18.0.1.2.0/pre-migration.py

```diff
@@ -0,0 +1,10 @@
+import logging
+
+from odoo.upgrade import util
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    _logger.info("Installing dependent module 'partner_type_base' by migration script.")
+    util.force_install_module(cr, "partner_type_base")
```

Comment:
Why do you need this? Odoo's ORM is in charge of that.

Reply:
I thought that too, but @houssine78 informed me that there is a bug and from my testing I can confirm that the added dependency is not installed automatically. https://github.com/odoo/odoo/issues/72703

Reply:
not totally as sbidoul wrote at the end of the ticket I've linked:

`Half fixed to be precise.`